### PR TITLE
CIF-1103 - Extend asset metadata editor with product field

### PR DESCRIPTION
**Pull request by LSantha**
Add new Commerce tab with cifproductfield to asset metadata editor.

## Description

As an author I want to tag assets with product SKU so that later I can easily find the assets related to that product .

The asset metadata editor should be extended with a new cifproductfield, the product SKUs from that field will be saved to cq:productSku property of the metadata node with type String[].

## Related Issue

CIF-1103

## How Has This Been Tested?

Manually.
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [X] All unit tests pass on CircleCi.
- [X] I ran all tests locally and they pass.

### CODE REVIEW

**Comment by herzog31** [comment]
This needs some kind of integration test, ideally a UI E2E test. We're touching the UI of a different team here, so we need to ensure that their changes do not break this feature and also to make sure it works with all supported AEM versions.

**Comment by LSantha** [comment]
@herzog31 , that's why I didn't request review then.

**Review by dplaton — APPROVED** [review]

File: content/cif-connector/src/main/content/jcr_root/libs/dam/content/schemaeditors/forms/default/image/items/tabs/items/tab4/.content.xml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<jcr:root xmlns:sling="http://sling.apache.org/jcr/sling/1.0" xmlns:granite="http://www.adobe.com/jcr/granite/1.0" xmlns:jcr="http://www.jcp.org/jcr/1.0" xmlns:nt="http://www.jcp.org/jcr/nt/1.0"
+    jcr:primaryType="nt:unstructured"
+    jcr:title="Commerce"
+    sling:resourceType="granite/ui/components/coral/foundation/container"
+    granite:rel="aem-assets-metadata-form-tab"
+    granite:class="cq-dam-asset-metadata-commerce"
+    listOrder="3">
+    <items jcr:primaryType="nt:unstructured">
+        <col1
```
Comment (mhaack) [inline content/cif-connector/src/main/content/jcr_root/libs/dam/content/schemaeditors/forms/default/image/items/tabs/items/tab4/.content.xml]: Is it best practice to add col1 & col3 as empty columns? Why not just put the product field in column 1?

**Comment by mhaack** [comment]
Generall question: why do we add `tab4` here? In my AEM instance there is already `/libs/dam/content/schemaeditors/forms/default/image/items/tabs/items/tab3` called "Product Data" why not just using this one?

**Comment by mhaack** [comment]
Additional question: does this also add the product field to the "Build Form" list in the side panel? If one builds his own asset metadata profile and would like the product field.

File: content/cif-connector/src/main/content/jcr_root/libs/dam/content/schemaeditors/forms/default/image/items/tabs/items/tab4/.content.xml (hunk 1 above)
Comment (LSantha) [reply content/cif-connector/src/main/content/jcr_root/libs/dam/content/schemaeditors/forms/default/image/items/tabs/items/tab4/.content.xml]: Indeed, that seems to be useless, although in the metadata editor they prefer to have the 3 columns even if they are empty.

**Comment by LSantha** [comment]
@mhaack, "Product Data" is old stuff and it has a number of fields when the asset is linked to a product. 
I thought to keep this feature apart from that and make it easily accessible on a dedicated tab.

**Comment by LSantha** [comment]
@mhaack , I'm not sure if I get your "additional question". Could you clarify it a bit?

**Comment by mhaack** [comment]
On the second question, the PR change adds the product field to the default asset metadata config. But there are customers with their own metadata config, so we need to add the product field also to the "Build Form" in the metadata editor (see http://localhost:4502/mnt/overlay/dam/gui/content/metadataschemaeditor/schemadetails.html/default?formPath=/conf/global/settings/dam/adminui-extension/metadataschema&pickable=)

**Comment by LSantha** [comment]
@mhaack, I see what you mean. The product field should have a well defined name mandated by this feature. So there is not much room for customization there. I also see that certain parts of the metadata are locked in the metadata editor. 
Now I'm thinking about making the product field compulsory instead of exposing it on the "Build Form" tab. WDYT?

**Comment by mhaack** [comment]
How would making it compulsory solve the problem if custom metadata profiles? Will they include the new tab always and by default?

File: content/cif-connector/src/main/content/jcr_root/libs/dam/content/schemaeditors/forms/default/image/items/tabs/items/tab4/.content.xml (hunk 1 above)
Comment (LSantha) [reply content/cif-connector/src/main/content/jcr_root/libs/dam/content/schemaeditors/forms/default/image/items/tabs/items/tab4/.content.xml]: @mhaack , BTW, the 3 columns are also related to the metadata schema editor, which adds the empty columns by default when you add a new tab. So, it's better to keep the empty columns.

Comment (mhaack) [reply content/cif-connector/src/main/content/jcr_root/libs/dam/content/schemaeditors/forms/default/image/items/tabs/items/tab4/.content.xml]: Ok that makes sense than.

**Comment by LSantha** [comment]
@mhaack , that's what I have to check.

**Comment by LSantha** [comment]
@mhaack , I've looked more into the asset metadata editor and asset metadataschema editor and I moved the extension from image metadata to the default metadata so that the extension would work for other asset types too like video.
I've also found that the "Build Form" part of the schema editor is hardwired in a large JSP that doesn't seem to be designed for extension by customers at all. Furthermore the extension being part of the default metadata it became part of the locked part of the metadata in the editor and the customers can add other fields to the metadata of various asset types but they cannot remove thid extension for a specific asset type. So I don't see a strong need for the product field to be available under the "Build Form" tab.

**Comment by mhaack** [comment]
> I moved the extension from image metadata to the default metadata so that the extension would work for other asset types too like video.
👍 

> I've also found that the "Build Form" part of the schema editor is hardwired in a large JSP that doesn't seem to be designed for extension by customers at all. 
This is about us providing the product field as an option here, not about customer extensions. Customers would be the user of that.

> So I don't see a strong need for the product field to be available under the "Build Form" tab.
I have no data on that, but I think for every customer profile this would be needed. Not sure how this is used by customers? Think our PM should have some opinion on that.

**Pull request merged**